frontend/src/components/RecordValidation.tsx

```diff
@@ -0,0 +1,246 @@
+      </Modal.Header>
+      <Modal.Body>
+        <div className={`${styles.tableHeight} ag-theme-alpine`}>
+          <AgGridReact<StatusItem>
+            groupDisplayType="groupRows"
+            rowData={validationDataForAgGrid}
+            columnDefs={validationColDefs}
+            defaultColDef={defaultColDef}
+            onGridReady={(params) => params.api.sizeColumnsToFit()}
+          />
+        </div>
+      </Modal.Body>
+      <Modal.Footer>
+        <Button onClick={onHide}>Close</Button>
+      </Modal.Footer>
+    </Modal>
+  );
+}
+
+function parseValidationReport(validationReport: string): Map<string, string> {
+  const validationReportMap = new Map<string, string>();
+  try {
+    // Parse the validation report JSON string
+    // e.g. "{"fastQs":"missing","igoComplete":"false"}"
+    return new Map(Object.entries(JSON.parse(validationReport)));
+  } catch (e) {
+    // Parse the alternative format of the validation report data
+    // e.g. "{fastQs=missing,igoComplete=false}"
+    const keyValuePairs = validationReport.replace(/[{}]/g, "").split(",");
+    for (const keyValuePair of keyValuePairs) {
+      const [key, value] = keyValuePair.split("=").map((str) => str.trim());
+      if (key && value) {
+        validationReportMap.set(key, value);
+      }
+    }
+    return validationReportMap;
+  }
+}
+
+function parseNestedSamplesValidationReport(input: string) {
+  const result: Array<[string, keyof typeof SAMPLE_STATUS_MAP]> = [];
+  try {
+    // Parse the validation report string as JSON
+    // e.g. "{"samples":[{"igoId":"06000_PD_1","cmoInfoIgoId":"06000_PD_1",...}]}"
+    const obj = JSON.parse(input);
+    if (obj && Array.isArray(obj.samples)) {
+      for (const sample of obj.samples) {
+        const igoId = sample.igoId;
+        const validationReport = sample?.status?.validationReport;
+        if (igoId && validationReport !== undefined) {
+          const reportMap =
+            validationReport instanceof Map
+              ? validationReport
+              : parseValidationReport(validationReport);
+          reportMap.forEach((value, key) => {
+            result.push([
+              igoId,
+              `${key} ${value}` as keyof typeof SAMPLE_STATUS_MAP,
+            ]);
+          });
+        }
+      }
+    }
+  } catch {
+    // Parse the alternative format of the validation report data
+    // e.g. "{samples=[{igoId=06000_PD_1, cmoInfoIgoId=06000_PD_1, ...}]}"
+    const pattern = /igoId=([^,}\s]+)[\s\S]*?validationReport=\{([^}]*)\}/g;
+    let match: RegExpExecArray | null;
+    while ((match = pattern.exec(input)) !== null) {
+      const igoId = match[1];
+      const reportContent = match[2];
+      const reportMap = parseValidationReport(`{${reportContent}}`);
+      reportMap.forEach((value, key) => {
+        result.push([
+          igoId,
+          `${key} ${value}` as keyof typeof SAMPLE_STATUS_MAP,
+        ]);
+      });
+    }
+  }
+  return result;
+}
```

Comment:
When discussing with Ben yesterday, he suggested that we should just fix this alternative format (e.g. "{fastQs=missing,igoComplete=false}"). No one is using request validation report data today, so we have the flexibility to freely experiment with cleaning it up. What do you think?

Reply:
100% that's the correct thing to do. I also 100% do not look forward to this task. Fixing broken JSON like that is such a headache but yes, absolutely needs to be done.